packages/design-system/src/scalars/components/toggle/toggle.tsx

```diff
@@ -1,28 +1,53 @@
-"use client";
-
 import * as React from "react";
 import * as SwitchPrimitives from "@radix-ui/react-switch";
 import { cn } from "@/scalars/lib/utils";
 
+interface ToggleProps {
+  disabled?: boolean;
+  checked?: boolean;
+  required?: boolean;
+  onChange?: (checked: boolean) => void;
+}
+
 const Toggle = React.forwardRef<
   React.ElementRef<typeof SwitchPrimitives.Root>,
-  React.ComponentPropsWithoutRef<typeof SwitchPrimitives.Root>
->(({ className, ...props }, ref) => (
-  <SwitchPrimitives.Root
-    className={cn(
-      "peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input",
+  React.ComponentPropsWithoutRef<typeof SwitchPrimitives.Root> & ToggleProps
+>(
+  (
+    {
       className,
-    )}
-    {...props}
-    ref={ref}
-  >
-    <SwitchPrimitives.Thumb
+      disabled = false,
+      checked = true,
+      required = false,
+      onChange = () => {},
+      ...props
+    },
+    ref,
+  ) => (
+    <SwitchPrimitives.Root
+      required={required}
+      checked={checked}
+      role="switch"
+      name="switch"
```

Comment:
the name should be a prop instead a hardcoded value

Reply:
you are right, done